### PR TITLE
Completion function signature

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/utils/expressions.py [hunk 1]
```diff
@@ -28,10 +28,31 @@ def get_function_listing():
     global listing
 
     if listing is None:
-        listing = [{'name': f['name'], 'display': f['description']} for f in DEFAULT_FUNCTION_MANAGER.build_listing()]
+        listing = [{'name': f['name'], 'display': f['description'], 'example': _build_function_signature(f)} for f in DEFAULT_FUNCTION_MANAGER.build_listing()]
     return listing
 
 
+def _build_function_signature(f):
```
Comment (rowanseymour) [inline temba/utils/expressions.py]: Unit tests for this

File: temba/utils/expressions.py [hunk 2]
```diff
@@ -28,10 +28,31 @@ def get_function_listing():
     global listing
 
     if listing is None:
-        listing = [{'name': f['name'], 'display': f['description']} for f in DEFAULT_FUNCTION_MANAGER.build_listing()]
+        listing = [{'name': f['name'], 'display': f['description'], 'example': _build_function_signature(f)} for f in DEFAULT_FUNCTION_MANAGER.build_listing()]
```
Comment (rowanseymour) [inline temba/utils/expressions.py]: Call it `signature` since example is something else

**Comment by nicpottier** [comment]
This doesn't seem to do the right thing when completing within a function.. should be showing all contact variables here.. (tel etc...)

rapidpro - visually build interactive sms applications 2015-11-06 09-01-54

**Comment by norkans7** [comment]
Right, I added a fix for that

**Pull request merged**